packages/commerce-sdk-react/src/hooks/ShopperCustomers/cache.ts

```diff
@@ -134,9 +135,37 @@ export const cacheUpdateMatrix: CacheUpdateMatrix<Client> = {
     },
     updateCustomerPassword: noop,
     updateCustomerProductList: TODO('updateCustomerProductList'),
-    updateCustomerProductListItem(customerId, {parameters}) {
+    updateCustomerProductListItem(customerId, {parameters}, response) {
+        const {itemId, listId} = parameters
         return {
-            update: [{queryKey: getCustomerProductListItem.queryKey(parameters)}],
+            update: [
+                {queryKey: getCustomerProductListItem.queryKey(parameters)},
+                // TODO: make separate PR that merges into develop
+                {
+                    // Also selectively update the lists
+                    queryKey: getCustomerProductLists.queryKey(parameters),
+                    updater: (oldData: ShopperCustomersTypes.CustomerProductListResult) => {
+                        const clone: ShopperCustomersTypes.CustomerProductListResult = JSON.parse(
+                            JSON.stringify(oldData)
+                        )
```

Comment:
JSON.parse has performance issues...
```suggestion
                        const clone: ShopperCustomersTypes.CustomerProductListResult = {...oldData}
```

Reply:
I had `{...oldData}` before, but it was just a shallow clone. I wanted a deep clone, since the property to update is nested.

Reply:
Here's the most concise (but maybe not the most readable) way to implement this logic. General idea is "Shallow clone the object, deep clone the array, but replace the array item with matching ID" for both the list of lists and for the list of items.
```ts
return {
  ...oldData,
  data: oldData.data.map((list) =>
    list.id !== listId
      ? list
      : {
        ...list,
        customerProductListItems:
          list.customerProductListItems?.map((item) =>
            item.id !== itemId ? item : response
          )
        }
  )
}
```

Reply:
❤️ Thanks Will! Not only it's more concise, but I think it's more readable too.